engine/execution/ingestion/engine.go

```diff
@@ -638,6 +642,16 @@ func (e *Engine) executeBlock(ctx context.Context, executableBlock *entity.Execu
 	if err != nil {
 		e.log.Err(err).Msg("failed in process block's children")
 	}
+
+	err, ok := <-computationResult.ExecutionDataProvideJob.Done
```

Comment:
1. I don't think it actually causes block execution to halt, because block execution happens in a separate gorouting for each block
2. We can just make the storing of the execution happen synchronously to simplify things.